### ISSUE TITLE
Add Astarte Device Rust SDK

### ISSUE THREAD

**Issue by harlem88** [astarte-platform/astarte-message-hub#5]
- [x] https://github.com/astarte-platform/astarte-message-hub/issues/27
- [x] https://github.com/astarte-platform/astarte-message-hub/issues/28

**Comment by harlem88** [comment]
Example of Astarte traits for wrapping Astarte SDK Device
``` rust
#[async_trait]
pub trait AstartePublisher: Send + Sync {
    async fn send_object<T: 'static>(
        &self,
        interface_name: &str,
        interface_path: &str,
        data: T,
    ) -> Result<(), AstarteError>
    where
        T: serde::Serialize + Send;
    async fn send(
        &self,
        interface_name: &str,
        interface_path: &str,
        data: AstarteType,
    ) -> Result<(), AstarteError>;
}

#[async_trait]
pub trait AstarteSubscriber {
    async fn subscribe(
        &self,
        astarte_node: &AstarteNode,
    ) -> Result<Receiver<Result<AstarteMessage, Status>>, AstarteError>;
    async fn unsubscribe(&self, astarte_node: AstarteNode) -> Result<(), AstarteError>;
}
```
https://github.com/harlem88/astarte-message-hub/blob/5d7268c94e785d44828dfeefdf8f23385b474dd4/src/data/mod.rs#L32

**Comment by FrancescoVaiani** [comment]
The first task is closed by #26
